ppl/src/main/java/org/opensearch/sql/ppl/PPLService.java

```diff
@@ -95,4 +108,20 @@ private PhysicalPlan plan(PPLQueryRequest request) {
         .plan(logicalPlan);
   }
 
+  private void setConnector(UnresolvedPlan unresolvedPlan) {
+    while (unresolvedPlan != null) {
```

Comment:
using Visitor pattern to traverse the tree?

Reply:
Yeah, my bad will look into it. Took a short cut as I have to only look into the root node for catalog.